internal/olm/operator/registry/operator_installer.go

```diff
@@ -122,49 +124,83 @@ func (o OperatorInstaller) waitForCatalogSource(ctx context.Context, cs *v1alpha
-			if og.GetName() == operator.SDKOperatorGroupName {
-				return fmt.Errorf("existing SDK-managed operator group's %s, "+
-					"please clean up existing operators `operator-sdk cleanup` before running package %q", msg, o.PackageName)
+
+	supported := o.SupportedInstallModes
+
+	if supported.Len() == 0 {
+		return nil, fmt.Errorf("operator %q is not installable: no supported install modes", o.StartingCSV)
+	}
+
+	// --install-mode was given
+	if !o.InstallMode.IsEmpty() {
+		// TODO: probably remove multinamespace
+		if o.InstallMode.InstallModeType == v1alpha1.InstallModeTypeSingleNamespace {
+			targetNsSet := sets.NewString(o.InstallMode.TargetNamespaces...)
+			if !supported.Has(string(v1alpha1.InstallModeTypeOwnNamespace)) && targetNsSet.Has(o.cfg.Namespace) {
+				return nil, fmt.Errorf("cannot watch namespace %q: operator %q does not support install mode %q", o.cfg.Namespace, o.StartingCSV, v1alpha1.InstallModeTypeOwnNamespace)
 			}
-			return fmt.Errorf("existing operator group %q's %s, "+
-				"please ensure it has the exact namespace set before running package %q", og.GetName(), msg, o.PackageName)
 		}
-		log.Infof("Using existing operator group %q", og.GetName())
-	} else {
-		// New SDK-managed OperatorGroup.
-		og = newSDKOperatorGroup(o.cfg.Namespace,
-			withTargetNamespaces(targetNamespaces...))
-		if err = o.cfg.Client.Create(ctx, og); err != nil {
-			return fmt.Errorf("error creating OperatorGroup: %w", err)
+		if o.InstallMode.InstallModeType == v1alpha1.InstallModeTypeSingleNamespace &&
+			o.InstallMode.TargetNamespaces[0] == o.cfg.Namespace {
+			return nil, fmt.Errorf("use install mode %q to watch operator's namespace %q", v1alpha1.InstallModeTypeOwnNamespace, o.cfg.Namespace)
+		}
+
+		supported = supported.Intersection(sets.NewString(string(o.InstallMode.InstallModeType)))
+		if supported.Len() == 0 {
+			return nil, fmt.Errorf("operator %q does not support install mode %q", o.StartingCSV, o.InstallMode.InstallModeType)
 		}
-		log.Infof("Created OperatorGroup: %s", og.GetName())
 
 	}
-	return nil
+	if !ogFound {
+		targetNamespaces, err := o.getTargetNamespaces(supported)
+		if err != nil {
+			return nil, err
+		}
+		if og, err = o.createOperatorGroup(ctx, targetNamespaces); err != nil {
+			return nil, fmt.Errorf("create operator group: %v", err)
+		}
+		log.Infof("operatorgroup %q created", og.Name)
+	} else if err := o.validateOperatorGroup(*og, supported); err != nil {
+		return nil, err
+	}
+
+	return og, nil
+}
+
+func (o *OperatorInstaller) createOperatorGroup(ctx context.Context, targetNamespaces []string) (*v1.OperatorGroup, error) {
+	og := newSDKOperatorGroup(o.cfg.Namespace, withTargetNamespaces(targetNamespaces...))
+	if err := o.cfg.Client.Create(ctx, og); err != nil {
+		return nil, err
+	}
+	return og, nil
+}
+
+func (o *OperatorInstaller) validateOperatorGroup(og v1.OperatorGroup, supported sets.String) error {
```

Comment:
This is called only if an `OperatorGroup` was found in-cluster right? So if there are any errors, they'll exist in the status. We shouldn't be re-validating an `OperatorGroup` with home-grown logic when OLM already does this; we only care about the target namespaces vs. install mode that we're creating an `OperatorGroup` with.

Reply:
changed `validateOperatorGroup` to `isOperatorGroupCompatible` and it is simpler.